src/wallet/api/schemas/walletApi.ts

```diff
@@ -0,0 +1,508 @@
+export const balanceResponseSchema = baseResponseSchema.extend({
+  balances: z.array(getBalanceObjectSchema),
+});
+
+/**
+ * Schema for transaction proposal inputs.
+ * Represents the inputs that will be used in a transaction.
+ */
+export const txProposalInputsSchema = z.object({
+  txId: z.string(),
+  index: z.number(),
+  addressPath: AddressPathSchema,
+});
+
+/**
+ * Schema for transaction proposal outputs.
+ * Represents the outputs that will be created in a transaction.
+ */
+export const txProposalOutputsSchema = z.object({
+  address: z.string(),
+  value: bigIntCoercibleSchema,
+  token: z.string(),
+  timelock: z.number().nullable(),
+});
+
+/**
+ * Response schema for creating a transaction proposal.
+ * Contains the proposal ID and the transaction details.
+ */
+export const txProposalCreateResponseSchema = baseResponseSchema.extend({
+  txProposalId: z.string(),
+  inputs: z.array(txProposalInputsSchema),
+});
+
+/**
+ * Response schema for updating a transaction proposal.
+ * Contains the proposal ID and the transaction hex.
+ */
+export const txProposalUpdateResponseSchema = baseResponseSchema.extend({
+  txProposalId: z.string(),
+  txHex: z.string(),
+});
+
+/**
+ * Schema for full node version data.
+ * Contains network parameters and configuration values.
+ * Uses passthrough() to allow additional fields in the response without breaking validation,
+ * as the full node may add new fields in future versions without changing the API version.
+ */
+export const fullNodeVersionDataSchema = z
+  .object({
+    timestamp: z.number(),
+    version: z.string(),
+    network: z.string(),
+    minWeight: z.number(),
+    minTxWeight: z.number(),
+    minTxWeightCoefficient: z.number(),
+    minTxWeightK: z.number(),
+    tokenDepositPercentage: z.number(),
+    rewardSpendMinBlocks: z.number(),
+    maxNumberInputs: z.number(),
+    maxNumberOutputs: z.number(),
+  })
+  .passthrough();
+
+/**
+ * Schema for full node transaction inputs.
+ * Represents the inputs of a transaction as seen by the full node.
+ */
+export const fullNodeInputSchema = z.object({
+  value: bigIntCoercibleSchema,
+  token_data: z.number(),
+  script: z.string(),
+  decoded: z.object({
+    type: z.string(),
+    address: z.string(),
+    timelock: z.number().nullable().optional(),
+    value: bigIntCoercibleSchema,
+    token_data: z.number(),
+  }),
+  tx_id: z.string(),
+  index: z.number(),
+  token: z.string().nullable().optional(),
+  spent_by: z.string().nullable().optional(),
+});
+
+/**
+ * Schema for full node transaction outputs.
+ * Represents the outputs of a transaction as seen by the full node.
+ */
+export const fullNodeOutputSchema = z.object({
+  value: bigIntCoercibleSchema,
+  token_data: z.number(),
+  script: z.string(),
+  decoded: z.object({
+    type: z.string(),
+    address: z.string(),
+    timelock: z.number().nullable().optional(),
+    value: bigIntCoercibleSchema,
+    token_data: z.number(),
+  }),
+  address: z.string(),
```

Comment:
Is it required even for data outputs? In that case, should you use the `AddressSchema`?

Reply:
It's not. Fixed and refactored all types to use `AddressSchema`